WooCommerce/StoreWidgets/StatsTimeRange.swift

```diff
@@ -25,4 +30,96 @@ extension StatsTimeRange {
             self = .thisYear
         }
     }
+
+    /// The maximum number of stats intervals a time range could have.
+    var maxNumberOfIntervals: Int {
+        switch self {
+        case .today:
+            return 24
+        case .thisWeek:
+            return 7
+        case .thisMonth:
+            return 31
+        case .thisYear:
+            return 12
+        }
+    }
```

Comment:
Do you think all of these properties should be shared in a commonplace with the main stats code?
Since we are going to update the analytics dashboard soon, there is a chance that these go outdated.

Reply:
I thought about moving `StatsTimeRangeV4` to `WooFoundation`, but it imports `Networking` models. Do you think it's ok to allow WooFoundation depend on Networking?

Original StatsTimeRangeV4 currently lives in Yosemite. I copy-pasted the code to prevent the import of `Yosemite`+`Storage` (+ `Networking` but it's already imported to widgets target).

Reply:
> WooFoundation depend on Networking
Probs not. Maybe we can move that to an extension helper that is imported in both targets?

Reply:
Can it be `WooFoundation`? I don't like idea of creating another library for a few models.
> move all related models to WF and make Networking import it?

- move `StatsTimeRangeV4` from Yosemite to WF
- move `StatGranularity` from Networking to WF
- import WF everywhere
- remove copy-pasted `StatsTimeRange` and use `StatsTimeRangeV4` in widgets from WF

Reply:
That also work, may involve more work tho!

Reply:
Then I'll merge current implementation into the feature branch and we can experiment with dependencies later.

Reply:
> there is a chance that these go outdated

☝️ What are the chances of this happening? What is the risk? How many times is this duplicated? 

Maybe we can also use the [Rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) as a guideline. 

Another library is also fine to me, assuming the cost is smaller compared to trying to place it in `WooFoundation`.